bokeh/client/session.py

```diff
@@ -153,11 +153,36 @@ def __init__(self, session_id=DEFAULT_SESSION_ID, io_loop=None, url=DEFAULT_SERV
         self._connection = ClientConnection(session=self, io_loop=io_loop, url=url)
 
         self._current_patch = None
+        self._callbacks = {}
 
     def _attach_document(self, document):
         self._document = document
         self._document.on_change(self._document_changed)
 
+    def add_periodic_callback(self, cb, callback_time):
+        from tornado import ioloop
+        self._callbacks.append(ioloop.PeriodicCallback(cb, callback_time, io_loop=self._connection._loop))
+
+    def add_periodic_callback(self, callback):
+        ''' Add callback so it can be invoked on a session periodically accordingly to period.
+
+        NOTE: periodic callbacks can only work within a session. It'll take no effect when bokeh output is html or notebook
+
+        '''
+        from tornado import ioloop
+        cb = self._callbacks[callback.id] = ioloop.PeriodicCallback(
+            callback.callback, callback.period, io_loop=self._connection._loop
+        )
+        cb.start()
+
+    def remove_periodic_callback(self, callback):
```

Comment:
the session add_periodic_callback/remove_periodic_callback methods are now private I think - maybe underscore-prefix `_add_periodic_callback` etc. ?